GSA_Adapter/GSAAdapter.cs

```diff
@@ -22,18 +22,48 @@
 
 using Interop.gsa_8_7;
 using BH.oM.GSA;
+using System.ComponentModel;
+using System;
+using BH.oM.Adapter;
+using BH.oM.Structure.Elements;
+using BH.Engine.Base.Objects;
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.MaterialFragments;
+using BH.oM.Structure.Constraints;
+using System.Collections.Generic;
+using BH.oM.Structure.SurfaceProperties;
 
 namespace BH.Adapter.GSA
 {
-    public partial class GSAAdapter : BHoMAdapter
+    public partial class GSAAdapter : StructuralAnalysisAdapter
     {
         /***************************************************/
         /**** Constructors                              ****/
         /***************************************************/
 
         public GSAAdapter(string filePath = "", GSAConfig gsaConfig = null, bool active = false)
         {
-            AdapterId = BH.Engine.GSA.Convert.AdapterID;
+            AdapterIdName = BH.Engine.GSA.Convert.AdapterIdName;
+
+            m_adapterComparers = new Dictionary<Type, object>
+            {
+                {typeof(Bar), new BH.Engine.Structure.BarEndNodesDistanceComparer(3) },
+                {typeof(Node), new BH.Engine.Structure.NodeDistanceComparer(3) },
+                {typeof(ISectionProperty), new BHoMObjectNameOrToStringComparer() },
+                {typeof(IMaterialFragment), new BHoMObjectNameComparer() },
+                {typeof(LinkConstraint), new BHoMObjectNameComparer() },
+            };
+
+            m_dependencyTypes = new Dictionary<Type, List<Type>>
+            {
+                {typeof(BH.oM.Structure.Loads.Load<Node>), new List<Type> { typeof(Node) }  },
+                {typeof(BH.oM.Structure.Loads.Load<Bar>), new List<Type> { typeof(Bar) } },
```

Comment:
Added the dependency between Loads and Nodes and Bars, so they can be pushed in the same Push.